### DISCUSSION TITLE
How to reset the Pitaya?

### DISCUSSION THREAD

**Discussion by LeakyEarth2000** [cmdwtf/Odiin#107] [Q&A]
Hey Chris! It's a been a very long time, but I am finally back to continue this project (from last year...)!

I have no idea what step I am up to, so I think it's best for me to reset my Pitaya Go. Problem is that none of the reset/user buttons actually work no longer. If I hold the user or reset button before plugging the power in, nothing happens! The green LED lights up, but no input. With my pitaya link, if I hold the only button (boot/reset) before plugging the power in directly to it, it will pop up a drive on my PC. with a firmware file. I tried putting it in DFU, but nothing happens. In fact, even if I plug in the pitaya go to my PC, my PC doesn't make that hardware connect sound and device manager doesn't acknowledge the device at all

As I said, I think it is best that I reset my Pitaya Go, but I have no idea how to now. I thought it would have been simple as holding the reset button down and plugging the USB C cable in! If you could guide me on what was the software needed in order to perform the reset, as well as the command to actually reset this thing, that would be greatly appreciated!

Cheers

**Comment by nitz** [comment]
Sounds like the bootloader is either missing or broken somehow. The easiest method for getting it to boot into the bootloader (if it's present) is to just double press the reset button. If you never see the LED turn on after doing that, then let's do this:

I think the best idea is to completely erase it and flash fresh bits via the Pitaya Link. I haven't used `pyocd` in several months, so I'm not sure what the most recent commands are, so you may have to adapt the ones I'm going to describe. Anyways, you should run something like `pyocd erase -t nrf52840 --chip` (It's the `erase` target in `Makefile` too, so `make erase` should do that.) That will completely blank the chip and ensure you're starting fresh.

Next, (assuming it's already built, go ahead and build it if its not,) you'll want to flash just the bootloader. (That's the `flash_bootloader` make target.) That'll only put the UF2 bootloader on it. From there, power cycling (or running `pyocd cmd -t nrf52840 --command reset`) should at least get the bootloader to run which will blink the red LED when it fails to find & load any application. You should be able to see it show up via USB in your device manager and such then, too.

Once that part seems alright, you can write a bit in the flash that will tell the bootloader that the "application" is valid and that it should try to launch it. That's the line in the makefile in the `flash_all` section that looks like `pyocd cmd -t nrf52840 --command write32 $(BOOTLOADER_BOOT_SETTING_ADDR) $(BOOTLOADER_BOOT_APP_VALID_VALUE) --command reset`, so with those values filled, you'll want to run `pyocd cmd -t nrf52840 --command write32 0xFF000 0x00000001 --command reset`.

After you do that, resetting the Pitaya Go, you should see the LED blink briefly, and then be off as it tries to load the application. Since there's no application actually there, basically nothing should happen at that point. (LED will be off, USB won't work, etc.) That means when you do flash the application, it should jump to it and start running.

If you ever need to get back to the bootloader, double-tapping the reset button should reset the board and boot into just the loader, preventing the loading of the application.

Also not super related, but I highly recommend [Uwe Siber's USB Tree View](https://www.uwe-sieber.de/usbtreeview_e.html). It's specifically for looking at connected USB devices, and I basically leave it open constantly when I'm working with devices like this, so I can see when they pop up and vanish and such. (It highlights devices green when they arrive and red when they remove, so it's easy to see changes!)

**Comment by LeakyEarth2000** [comment]
> Next, (assuming it's already built, go ahead and build it if its not

My apologises, but I can't seem to find nor remember how to build it. Could you please guide me how to re-build the software? Thank you!
